content/chainguard/administration/assumable-ids/identity-examples/jenkins-identity-chainctl/index.md

```diff
@@ -0,0 +1,129 @@
+---
+title : "Use chainctl to Create an Assumable Identity for a Jenkins Pipeline"
+linktitle: "Jenkins with chainctl"
+aliases:
+lead: ""
+description: "How to use chainctl to create a Chainguard identity that can be assumed by a Jenkins Pipeline."
+type: "article"
+date: 2025-08-27T08:03:42+00:00
+lastmod: 2025-08-27T08:03:42+00:00
+draft: false
+tags: ["Chainguard Containers", "Procedural"]
+images: []
+weight: 030
+---
+
+This guide explains how to configure Jenkins to authenticate with Chainguard using an assumable identity. This example uses `chainctl`, Chainguard's command line tool. If you would like to follow this guide using Terraform, you can review [Use Terraform to Create an Assumable Identity for a Jenkins Pipeline](/chainguard/administration/assumable-ids/identity-examples/jenkins-identity-terraform/).
+
+With this setup, Jenkins jobs can securely push and pull container images from the Chainguard registry without managing static credentials.
+
+
+## Prerequisites
+
+- A running Jenkins controller or agent with:
+    - chainctl installed and available in $PATH.
+    - Docker or another container runtime installed.
+- Admin access to your Chainguard organization.
+- An existing assumable ID created for Jenkins, or permissions to create one.
+
+
+## Create a Jenkins Assumable ID
+
+Use chainctl to create an identity that Jenkins can use to authenticate:
+
+```sh
+chainctl iam identities create \
```

Comment:
Off the bat I can't really test this guide, as this command won't work as written:

```
mark@mark-chainguard:~$ chainctl iam identities create  jenkins-ci   --description "Identity for Jenkins builds"   --output json
                                    
   ✔ Selected folder chainguard.edu.
{"error":"issuer or issuer pattern is required"}
```

Reply:
I think this procedure might be a more complicated than when the jenkins doc was first written. Jenkins doesn't seem to have oidc by default